src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs

```diff
@@ -530,12 +530,7 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<T> ExcludeFromManifest<T>(this IResourceBuilder<T> builder) where T : IResource
     {
-        foreach (var annotation in builder.Resource.Annotations.OfType<ManifestPublishingCallbackAnnotation>())
-        {
-            builder.Resource.Annotations.Remove(annotation);
-        }
-
-        return builder.WithAnnotation(ManifestPublishingCallbackAnnotation.Ignore);
+        return builder.WithManifestPublishingCallback(ManifestPublishingCallbackAnnotation.Ignore);
```

Comment:
Modified this to use the specific method we have. It takes care of setting the Append/Replace rules.

Reply:
Ignore that ;)